x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/configuration/ConfigurationValidation.java

```diff
@@ -75,6 +73,22 @@ public ConfigurationValidation(StreamInput in) throws IOException {
         );
     }
 
+    /**
+     * Parses the value of a constraint from the XContentParser stream.
+     * This method is designed to handle various types of constraint values as per the connector's protocol original specification.
+     * The constraints can be of type string, number, or list of values.
+     */
+    private static Object parseConstraintValue(XContentParser p) throws IOException {
+        if (p.currentToken() == XContentParser.Token.VALUE_STRING) {
+            return p.text();
+        } else if (p.currentToken() == XContentParser.Token.VALUE_NUMBER) {
+            return p.numberValue();
+        } else if (p.currentToken() == XContentParser.Token.START_ARRAY) {
+            return p.list();
```

Comment:
This is the extension to parser, we allow for list type of arbitrary values as per original connector protocol spec